scripts/paper/archive_supporting_web_evidence.py

```
@@ -0,0 +1,114 @@
+    ("zkpytorch_blog", "https://blog.polyhedra.network/zkpytorch/"),
+    ("paradex_stwo", "https://starkware.co/blog/paradex-s-two-the-fastest-prover-meets-the-fastest-appchain/"),
+]
+
+
+def slugify(text: str) -> str:
+    return re.sub(r"[^a-z0-9]+", "_", text.lower()).strip("_")
+
+
+def fetch(url: str) -> bytes:
+    try:
+        req = urllib.request.Request(url, headers={"User-Agent": "codex-paper-archiver"})
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)